redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/protocolHostGeneration.kt

```diff
@@ -251,23 +251,24 @@ internal class ProtocolButton<W : Any>(
   private val json: Json,
   private val mismatchHandler: ProtocolMismatchHandler,
 ) : ProtocolNode<W>(id, WidgetTag(4)) {
-  private var _widget: Button<W> = widget
-  override val widget: Widget<W> get() = _widget
+  private var _widget: Button<W>? = widget
+  override val widget: Widget<W> get() = _widget ?: error("detached")
 
   private val serializer_0: KSerializer<String?> = json.serializersModule.serializer()
   private val serializer_1: KSerializer<Boolean> = json.serializersModule.serializer()
 
   public override fun apply(change: PropertyChange, eventSink: EventSink): Unit {
+    val widget = this.widget
```

Comment:
Not that it matters all that much...
```suggestion
    val widget = _widget ?: error("detached")
```